src/FloorSubsystem/Floor.java

```diff
@@ -0,0 +1,56 @@
+package FloorSubsystem;
+import SchedulerSubsystem.Scheduler;
+import SchedulerSubsystem.SchedulerEvent;
+import Utlities.Passenger;
+import java.util.ArrayList;
+
+
+public class Floor implements Runnable {
+    private ArrayList<Passenger> passengers;
+    private int floorLamp;
+    private Scheduler scheduler;
+
+    public Floor(int floorNumber, Scheduler scheduler) {
+        this.passengers = new ArrayList<>();
+        this.floorLamp = floorNumber;
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * Receives notification from scheduler that the elevator floor has changed.
+     */
+    public void receiveElevatorFloorDisplayChangeRequest() {
```

Comment:
I think the scheduler can just update the floor with the current state of the elevator. The floor can react on its own model updates to decide to change the lamp.